cmd/zoekt-mirror-gerrit/main.go

```diff
@@ -224,3 +229,15 @@ func marshalBool(b bool) string {
 	}
 	return "0"
 }
+
+func anonymousURL(u *url.URL) string {
+	anon := *u
+	anon.User = nil
+	return anon.String()
+}
+
+func addPassword(u string, user *url.Userinfo) string {
+	password, _ := user.Password()
+	username := user.Username()
+	return strings.Replace(u, fmt.Sprintf("://%s@", username), fmt.Sprintf("://%s:%s@", username, password), 1)
+}
```

Comment:
This is prone to breaking, I suggest using url.URL
```suggestion
func addPassword(rawURL string, user *url.Userinfo) string {
    u, err := url.Parse(rawURL)
    if err != nil {
        return rawURL
    }
    u.User = user
    return u.String()
}
```

Reply:
The problem with this solution is that il will encode the path and `${project}` is transformed to `$%7Bproject%7D` and break all `strings.Replace(…, "${project}")`

I can replace `$%7Bproject%7D`  → `${project}` before `return` (but I'm not sure it's a good solution)
or I can use `$%7Bproject%7D`  everywhere in the file
(I'm not convinced by either solution)

(besides, I noticed that the management of paths (host + repo name) could surely be simplified to centralize part of the logic)

What do you think is the solution?

Reply:
Alright I see now why we avoided URL parsing before, its not really a URL but a URL template. I think we will stick with this then, but a potential better solution is to only add in the password at the time we create the cloneURL.